LibreNMS/OS/Teldat.php

```diff
@@ -0,0 +1,309 @@
+
+        $data = snmpwalk_cache_oid($this->getDeviceArray(), 'teldatCellularStateMobileRxRSRQ', [], 'TELDAT-MON-INTERF-CELLULAR-MIB');
+
+        if (! empty($data)) {
+            $ifNames = $this->getCacheByIndex('ifName', 'IF-MIB');
+            $ifOperStatuses = $this->getCacheByIndex('ifOperStatus', 'IF-MIB');
+
+            foreach ($data as $index => $entry) {
+                if (Str::startsWith($ifNames[$index], 'cellular') && $ifOperStatuses[$index] == 'up') {
+                    $sensors[] = new WirelessSensor(
+                        'rsrq',
+                        $this->getDeviceId(),
+                        '.1.3.6.1.4.1.2007.4.1.2.2.2.18.3.2.1.23.' . $index,
+                        'teldat',
+                        $index,
+                        'RSRQ: ' . $this->shortIfName($ifNames[$index], $index),
+                        $entry['teldatCellularStateMobileRxRSRQ']
+                    );
+                }
+            }
+        }
+
+        return $sensors;
+    }
+
+    /**
+     * teldatCellularStateMobileRxRSRP = .1.3.6.1.4.1.2007.4.1.2.2.2.18.3.2.1.22 = "Cellular mobile reference symbol received power (RSRP)."
+     *
+     * Discover wireless RSRP (Reference Signals Received Power). This is in dBm. Type is rsrp.
+     * Returns an array of LibreNMS\Device\Sensor objects that have been discovered
+     *
+     * @return array
+     */
+    public function discoverWirelessRsrp()
+    {
+        $sensors = [];
+
+        $data = snmpwalk_cache_oid($this->getDeviceArray(), 'teldatCellularStateMobileRxRSRP', [], 'TELDAT-MON-INTERF-CELLULAR-MIB');
+
+        if (! empty($data)) {
+            $ifNames = $this->getCacheByIndex('ifName', 'IF-MIB');
+            $ifOperStatuses = $this->getCacheByIndex('ifOperStatus', 'IF-MIB');
+
+            foreach ($data as $index => $entry) {
+                if (Str::startsWith($ifNames[$index], 'cellular') && $ifOperStatuses[$index] == 'up') {
+                    $sensors[] = new WirelessSensor(
+                        'rsrp',
+                        $this->getDeviceId(),
+                        '.1.3.6.1.4.1.2007.4.1.2.2.2.18.3.2.1.22.' . $index,
+                        'teldat',
+                        $index,
+                        'RSRP: ' . $this->shortIfName($ifNames[$index], $index),
+                        $entry['teldatCellularStateMobileRxRSRP']
+                    );
+                }
+            }
+        }
+
+        return $sensors;
+    }
+
+    /**
+     * teldatCellularStateMobileCellId    = .1.3.6.1.4.1.2007.4.1.2.2.2.18.3.2.1.5  = "Cellular mobile used cell id (+CGREG)."
+     * teldatCellularStateMobileLTECellId = .1.3.6.1.4.1.2007.4.1.2.2.2.18.3.2.1.25 = "Cellular mobile used LTE cell id."
+     *
+     * Discover wireless Cellular Cell Id. This is in cell number. Type is cellid.
+     * Returns an array of LibreNMS\Device\Sensor objects that have been discovered
+     *
+     * @return array
+     */
+    public function discoverWirelessCell()
```

Comment:
Cell information returns: CellID and LteCellID, this is usual in troubleshooting the device communication with the providers antenna, this may help to detect noise in some channels at particular dense areas with more than one cell tower.
But yeah, it's not for the most common management teams.

other meaningfull might be more important like 'teldatCellularStateMobileRegistrationState', 'temperature' or even teldatCellularStateMobile (Rx/Tx) BpsLast (X) s ... but i didn't find a metric type in LibreNMS 'https://docs.librenms.org/Developing/os/Wireless-Sensors/' that I might adapt to this in a straightforward way. I'm still taking baby steps with LibreNMS...

But, since I've cloned CiscoCellular and there was the Cell information, I actually just kept true to the original form, so "discoverWirelessCell" came from CiscoCellular.php, I just adapted to Teldat information, the best way I could...

Reply:
I still bellive that Cellular should be a Module apart from Wireless...

Who knows If I could help you guys to do more in depth work, like that, in the future...
But still I need to earn trust in myself and my PR's... not sure if I'm ready for that task.

Reply:
@murrant  , I've detected a possible issue here:

![image](https://user-images.githubusercontent.com/12242582/134394852-b824889b-8656-4edc-b78d-771fa71f174c.png)

Reply:
teldatCellularStateMobileRegistrationState and teldatCellularStateMobileRadioTechnology are probably state sensors and teldatCellularStateMobileTemperature is a temperature sensor.

Bands are often converted to a frequency.

Reply:
ok, thx.